src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -70,26 +68,32 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     PreparedQuery results = datastore.prepare(query);
     Entity userInfoEntity = results.asSingleEntity();
     if (userInfoEntity == null) {
-    response.setContentType("text/html;");
-    response.getWriter().println("You do not belong to a family yet! Get method");
-   // System.out.println("You do not belong to a family yet! Get method");
+        response.setContentType("text/html;");
+        response.getWriter().println("You do not belong to a family yet!");
     return;
     }
-
+    
     // creates arraylist and starts query
     ArrayList<String> groceryList = new ArrayList<String>();
     Query groceryQuery = new Query("Grocery");
```

Comment:
One suggestion is instead of getting all of the grocery items and then going through and checking the familly ID, you could modify the query to only return grocery items with the correct familyID property - 

Query query = new Query("Grocery")
        .setFilter(new Query.FilterPredicate("familyID", Query.FilterOperator.EQUAL, familyID));

Reply:
+1. Iterating through all the grocery items is not scalable.

Reply:
solved